ddht/v5_1/client.py

```diff
@@ -437,3 +463,111 @@ async def topic_query(
         request_id: Optional[bytes] = None,
     ) -> InboundMessage[FoundNodesMessage]:
         raise NotImplementedError
+
+
+@asynccontextmanager
+async def common_client_stream_find_nodes(
+    client: Client,
+    node_id: NodeID,
+    endpoint: Endpoint,
+    distances: Collection[int],
+    *,
+    request_id: Optional[bytes] = None,
+) -> AsyncIterator[trio.abc.ReceiveChannel[InboundMessage[FoundNodesMessage]]]:
+    async def _stream_find_nodes_response(
+        client: Client,
```

Comment:
you can leave this out of the function args since it is available within the containing context.

Reply:
Ah, yes of course and this also takes care of the mypy complaining about too many arguments for `nursery.start_soon()`!